include/RAJA/util/TypedViewBase.hpp

```diff
@@ -234,12 +234,24 @@ class ViewBase {
 
   public:
 
-    constexpr ViewBase() = default;
-    RAJA_INLINE constexpr ViewBase(ViewBase const &) = default;
-    RAJA_INLINE constexpr ViewBase(ViewBase &&) = default;
-    RAJA_INLINE ViewBase& operator=(ViewBase const &) = default;
-    RAJA_INLINE ViewBase& operator=(ViewBase &&) = default;
 
+    RAJA_HOST_DEVICE
```

Comment:
Do you think it's worth guarding this so that the non-default constructors are only used with nvcc?

Reply:
I recall that the hip compiler had a different set of preferences for defaulted methods. Also, I think compilers that support openmp target offload prefer defaulted ctors, copy/move ctors, etc. since that is needed to make objects trivially copyable so map clauses do the right thing and you don't get a bunch of compiler warning spew. As long as it does the right thing for these cases, it works for me.

Reply:
yes.  after talking to @rhornung67 I think that would be the best... I'll make some changes.

Reply:
Yes, it should be defaulted any time it can be.  The target offload compilers sometimes prefer it, but more to the point it can make some versions of these immensely cheaper to compile, instantiate and copy by turning them into essentially POD types when the contents are PODs.  It makes me wonder what is causing the failure, since I thought we ironed out the 10.1 failures for these and it could be caused by interactions with chai types.  Is there a corresponding CHAI PR?